libr/anal/sign.c

```diff
@@ -681,6 +864,44 @@ R_API void r_sign_list(RAnal *a, int format) {
 	}
 }
 
+R_API char *r_sign_calc_bbhash(RAnal *a, RAnalFunction *fcn) {
+	RListIter *iter = NULL;
+	RAnalBlock *bbi = NULL;
+	char *digest_hex = NULL;
+	RCore *core = a->coreb.core;
+	RHash *ctx = r_hash_new (true, R_ZIGN_HASH);
+	r_hash_do_begin (ctx, R_ZIGN_HASH);
+	r_list_foreach (fcn->bbs, iter, bbi) {
```

Comment:
i think bbs are sorted by offset. we can probably add an r_list_sort before thst foreach to ensure this